File: app/src/main/java/net/sarasarasa/basemvp/mvp/ui/MainActivity.java
package net.sarasarasa.basemvp.mvp.ui;

import android.os.Bundle;

import net.sarasarasa.basemvp.R;
import net.sarasarasa.basemvp.base.MvpActivity;
import net.sarasarasa.basemvp.mvp.contract.MainContract;
import net.sarasarasa.basemvp.mvp.presenter.MainPresenter;

public class MainActivity extends MvpActivity<MainContract.Presenter> implements MainContract.View {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        layoutResID = R.layout.activity_main;
        super.onCreate(savedInstanceState);
    }

    @Override
    protected MainContract.Presenter createPresenter() {
        return new MainPresenter();
    }

    @Override
    protected void bindView() {

    }

    @Override
    protected void initView() {

    }

}

File: kotlin/src/main/java/net/sarasarasa/basemvpkotlin/mvp/ui/MainActivity.kt
package net.sarasarasa.basemvpkotlin.mvp.ui

import android.os.Bundle
import net.sarasarasa.basemvpkotlin.R
import net.sarasarasa.basemvpkotlin.base.MvpActivity
import net.sarasarasa.basemvpkotlin.mvp.contract.MainContract
import net.sarasarasa.basemvpkotlin.mvp.presenter.MainPresenter

class MainActivity : MvpActivity<MainContract.View, MainPresenter>(), MainContract.View {


    override fun onCreate(savedInstanceState: Bundle?) {
        layoutResID = R.layout.activity_main
        super.onCreate(savedInstanceState)
    }

    override fun initView() {

    }

    override fun bindView() {

    }

    override fun createPresenter(): MainPresenter {
        return MainPresenter()
    }

}
